src/tokens/slice.ts

```diff
@@ -100,6 +100,7 @@ export interface TokenBalancesWithAddress {
 
 export interface State {
   tokenBalances: StoredTokenBalances
+  importedTokenIds: string[]
```

Comment:
ah, i see where the confusion comes from. since the main focus for this task is the token lookup, i think it's probably sensible to defer the token already imported error until the next task. you can add a TODO in the code. once the tokens are saved, it is only a few more lines of code to display the error (compared to if you do it now, to get the error you basically need to implement the whole token saving part).

we should avoid storing derived state in redux (same as any other data source) - we'll need to eventually be able to store the token info and display it in some screens, so we'll need more than just the token id. we can create selectors to get a list of token id's that are imported, but we should not have a specific state that tracks the token id's in this case